tests/lts_compatibility.py

```diff
@@ -0,0 +1,253 @@
+            major_version = Version(version).release[0]
+            # Explictly test the logging app as it was packaged in the very first LTS
+            # TODO: Is this the right approach?
+            args.js_app_bundle = os.path.join(lts_install_path, "samples/logging/js")
+        else:
+            version = args.ccf_version
+            binary_dir = LOCAL_CHECKOUT_DIRECTORY
+            library_dir = LOCAL_CHECKOUT_DIRECTORY
+            major_version = None
+
+        network_args = {
+            "hosts": args.nodes,
+            "binary_dir": binary_dir,
+            "library_dir": library_dir,
+            "txs": txs,
+            "version": major_version,
+        }
+        if idx == 0:
+            LOG.info(f"Starting new service (version: {version})")
+            network = infra.network.Network(**network_args)
+            network.start_and_join(args)
+        else:
+            LOG.info(f"Recovering service (new version: {version})")
+            network = infra.network.Network(**network_args, existing_network=network)
+            network.start_in_recovery(
+                args,
+                ledger_dir,
+                committed_ledger_dir,
+                snapshot_dir=snapshot_dir,
+            )
+            network.recover(args)
+
+        nodes = network.get_joined_nodes()
+        primary, _ = network.find_primary()
+
+        # Verify that all nodes run the expected CCF version
+        if not major_version or major_version > 1:
+            for node in nodes:
+                with node.client() as c:
+                    r = c.get("/node/version")
+                    assert (
+                        r.body.json()["ccf_version"] == version
+                    ), f"Node version is not {version}"
+
+        issue_activity_on_live_service(network, args)
+
+        snapshot_dir = (
+            network.get_committed_snapshots(primary) if use_snapshot else None
+        )
+
+        network.stop_all_nodes(verbose_verification=False)
+        ledger_dir, committed_ledger_dir = primary.get_ledger(
+            include_read_only_dirs=True
+        )
+
+        # Check that the ledger can be parsed on all nodes
+        for node in nodes:
+            ccf.ledger.Ledger(node.remote.ledger_paths()).get_latest_public_state()
+
+
+if __name__ == "__main__":
+
+    def add(parser):
+        parser.add_argument(
+            "--latest-lts-file",
+            help="File containing the latest LTS",
+            type=str,
+        )
+
+    args = infra.e2e_args.cli_args(add)
+
+    # JS generic is the only app included in CCF install
+    args.package = "libjs_generic"
+    args.nodes = infra.e2e_args.max_nodes(args, f=0)
+
+    # Hardcoded because host only accepts from info on release builds
+    args.host_log_level = "info"
+
+    run_live_compatibility_since_last(args)
+    run_ledger_compatibility_since_first(args, use_snapshot=False)
+    run_ledger_compatibility_since_first(args, use_snapshot=True)
```

Comment:
Running this test twice with and without snapshots seems wasteful. We could randomly alternate between recovery from snapshot/without snapshot in the test instead?

Reply:
It depends. In the main Daily, yes. On a release branch when we run once before release, we probably want to be exhaustive if we can.

Reply:
Agreed. I'll use our cmake's `LONG_TESTS` flag to indicate this.